test/HealthChecks.AzureServiceBus.Tests/AzureServiceBusQueueMessageCountThresholdHealthCheckTests.cs

```diff
@@ -0,0 +1,290 @@
+        var otherContext = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HealthCheckName, healthCheck, HealthStatus.Unhealthy, null)
+        };
+
+        var otherActual = await healthCheck
+            .CheckHealthAsync(context, tokenSource.Token)
+            .ConfigureAwait(false);
+
+        _clientProvider
+            .Received(1)
+            .CreateManagementClient(ConnectionString);
+    }
+
+    [Fact]
+    public async Task can_create_client_with_fully_qualified_endpoint()
+    {
+        using var tokenSource = new CancellationTokenSource();
+        var options = new AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions(TopicName)
+        {
+            FullyQualifiedNamespace = FullyQualifiedName,
+            Credential = _tokenCredential,
+        };
+        var healthCheck = new AzureServiceBusQueueMessageCountThresholdHealthCheck(options, _clientProvider);
+        var context = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HealthCheckName, healthCheck, HealthStatus.Unhealthy, null)
+        };
+
+        var actual = await healthCheck
+            .CheckHealthAsync(context, tokenSource.Token)
+            .ConfigureAwait(false);
+
+        _clientProvider
+            .Received(1)
+            .CreateManagementClient(options.FullyQualifiedNamespace, options.Credential);
+    }
+
+    [Fact]
+    public async Task reuses_existing_client_when_using_same_fully_qualified_name_with_different_topic()
+    {
+        // First call
+        using var tokenSource = new CancellationTokenSource();
+        var options = new AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions(TopicName)
+        {
+            FullyQualifiedNamespace = FullyQualifiedName,
+            Credential = _tokenCredential,
+        };
+        var healthCheck = new AzureServiceBusQueueMessageCountThresholdHealthCheck(options, _clientProvider);
+        var context = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HealthCheckName, healthCheck, HealthStatus.Unhealthy, null)
+        };
+
+        var actual = await healthCheck
+            .CheckHealthAsync(context, tokenSource.Token)
+            .ConfigureAwait(false);
+
+        // Second call
+        var otherTopicName = Guid.NewGuid().ToString();
+        var otherOptions = new AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions(otherTopicName)
+        {
+            FullyQualifiedNamespace = FullyQualifiedName,
+            Credential = _tokenCredential,
+        };
+        var otherHealthCheck = new AzureServiceBusQueueMessageCountThresholdHealthCheck(options, _clientProvider);
+        var otherContext = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HealthCheckName, healthCheck, HealthStatus.Unhealthy, null)
```

Comment:
```suggestion
            Registration = new HealthCheckRegistration(HealthCheckName, otherHealthCheck, HealthStatus.Unhealthy, null)
```

Please check all tests for such typos.